### ISSUE TITLE
[JENKINS-56913] Support for filtering all occurrences of all parallel() branches except one in the job log

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/workflow-job-plugin#685]
<p>As suggested by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> in <a href="https://issue-redirect.jenkins.io/issue/54304" title="Jobs in parallel dont display the variant they are running." class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/54304">JENKINS-54304</a>"><del>JENKINS-54304</del></a> for supporting the show/hide feature for all occurrences for a branch resp. for all branches except the desired one I open this ticket.</p>

<p>Feature request would be to be able to choose a <tt>parallel</tt> branch by it's name and hide any line in the console log which belongs to a different branch in order to quickly see the log for one branch only. The remaining job log not part of the parallel step shall (optionally?) be still displayed (blue ocean as well currently does not provide this).</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/macdrega.png" title="macdrega's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=macdrega">macdrega</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56913" target="_blank">Support for filtering all occurrences of all parallel() branches except one in the job log</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: workflow-job-plugin
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 3
<li><b>watchers</b>: 4
<li><b>imported</b>: 20251212-090250
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>As suggested by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> in <a href="https://issues.jenkins.io/browse/JENKINS-54304" title="Jobs in parallel dont display the variant they are running." class="issue-link" data-issue-key="JENKINS-54304"><del>JENKINS-54304</del></a> for supporting the show/hide feature for all occurrences for a branch resp. for all branches except the desired one I open this ticket.</p>

<p>Feature request would be to be able to choose a <tt>parallel</tt> branch by it's name and hide any line in the console log which belongs to a different branch in order to quickly see the log for one branch only. The remaining job log not part of the parallel step shall (optionally?) be still displayed (blue ocean as well currently does not provide this).</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56913?focusedId=364320&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-364320">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>A related idea (just throwing this out there): somewhere discoverable (for example next to <b>(show)</b> / <b>(hide)</b>), offer a link to a dedicated log URL for that branch, or more generally any block or non-block node. Such a URL does not currently exist in general—there are links for log text from a specific <tt>FlowNode</tt> (typically a step), but this does not include text from nested steps, so these displays are mostly useful for atom steps like <tt>sh</tt>.</p>

<p>Note that for efficient display (esp. with <tt>pipeline-cloudwatch-logs</tt>) this would deserve a new API method in <tt>LogStorage</tt>, since <tt>stepLog</tt> as mentioned above does not include text from descendants.</p>
</pre>
</details>

</ul>
</details>

<p>A related idea (just throwing this out there): somewhere discoverable (for example next to <b>(show)</b> / <b>(hide)</b>), offer a link to a dedicated log URL for that branch, or more generally any block or non-block node. Such a URL does not currently exist in general—there are links for log text from a specific <tt>FlowNode</tt> (typically a step), but this does not include text from nested steps, so these displays are mostly useful for atom steps like <tt>sh</tt>.</p>

<p>Note that for efficient display (esp. with <tt>pipeline-cloudwatch-logs</tt>) this would deserve a new API method in <tt>LogStorage</tt>, since <tt>stepLog</tt> as mentioned above does not include text from descendants.</p>
